src/Whim/Workspace/Workspace.cs

```diff
@@ -228,28 +231,43 @@ private void UpdateLayoutEngine(int delta)
 		);
 	}
 
+	private void CycleLayoutEngine(int delta)
+	{
+		int nextIdx;
+
+		lock (_workspaceLock)
+		{
+			nextIdx = (_activeLayoutEngineIndex + delta).Mod(_layoutEngines.Length);
+		}
+
+		SetLayoutEngine(nextIdx);
+	}
+
 	public void NextLayoutEngine()
 	{
 		Logger.Debug(Name);
-		UpdateLayoutEngine(1);
+		CycleLayoutEngine(1);
 	}
 
 	public void PreviousLayoutEngine()
 	{
 		Logger.Debug(Name);
-		UpdateLayoutEngine(-1);
+		CycleLayoutEngine(-1);
 	}
 
-	public bool TrySetLayoutEngine(string name)
+	public void LastActiveLayoutEngine()
+	{
+		SetLayoutEngine(_prevLayoutEngineIndex);
+	}
+
+	public bool SetLayoutEngineFromName(string name)
```

Comment:
I agree that this is a bit clearer, though I'd like to keep the `Try` prefix. 

I think that `SetLayoutEngine` and this should either both share the same name, or both be explicit (e.g., `TrySetLayoutEngineFromName`)